### PR TITLE
Switch postgres in the build container to postgresql14

### PR DESCRIPTION
**Pull request by danmilleruk**
Unfortunately this means we need to explicitly install some stuff.

Let me know if I've missed anything.

### CODE REVIEW

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,25 @@
-FROM public.ecr.aws/lambda/ruby:3.2
-
-# Install tools used when installing / building gems
-RUN yum install -y postgresql-devel sqlite-devel openssl-devel wget make gcc gcc-c++ git \
- && yum clean all \
- && rm -rf /var/cache/yum
+FROM amazonlinux:2023
```
Comment (choffee) [inline ci/aws-lambda-ruby-dev/3.2/Dockerfile]: can we use fully qualified here: eg docker.io/amazonlinux:2023
Makes it a little clearer but also works with podman  and other container engines. :)

**Comment by choffee** [comment]
I don't think this is going to work in the lambda.

Grabbing the lib we are interested in and running ldd in the lambda gives

```
bash-4.2# ldd libpq.so.private15-5.15 
./libpq.so.private15-5.15: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by ./libpq.so.private15-5.15)
./libpq.so.private15-5.15: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by ./libpq.so.private15-5.15)
	linux-vdso.so.1 (0x00007fff3a5da000)
	libssl.so.3 => not found
	libcrypto.so.3 => not found
	libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007fd44f800000)
	libldap_r-2.4.so.2 => /lib64/libldap_r-2.4.so.2 (0x00007fd44f400000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fd44f000000)
	libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007fd44ec00000)
	libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007fd44e800000)
	libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007fd44e400000)
	libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007fd44e000000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fd44dc00000)
	libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007fd44d800000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007fd44d400000)
	liblber-2.4.so.2 => /lib64/liblber-2.4.so.2 (0x00007fd44d000000)
	libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007fd44cc00000)
	libssl.so.10 => /lib64/libssl.so.10 (0x00007fd44c800000)
	libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007fd44c200000)
	libssl3.so => /lib64/libssl3.so (0x00007fd44be00000)
	libsmime3.so => /lib64/libsmime3.so (0x00007fd44ba00000)
	libnss3.so => /lib64/libnss3.so (0x00007fd44b600000)
	libnssutil3.so => /lib64/libnssutil3.so (0x00007fd44b200000)
	libplds4.so => /lib64/libplds4.so (0x00007fd44ae00000)
	libplc4.so => /lib64/libplc4.so (0x00007fd44aa00000)
	libnspr4.so => /lib64/libnspr4.so (0x00007fd44a600000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fd44a200000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fd44fc00000)
	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007fd449e00000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fd449a00000)
	libz.so.1 => /lib64/libz.so.1 (0x00007fd449600000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fd449200000)
	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007fd448e00000)
```

**Comment by danmilleruk** [comment]
Yeah, hold fire on doing anything with this. It builds, but its not going to work.

I'll rethink the strategy.

**Comment by choffee** [comment]
Yes this looks good and reflects the shared layer that we are using. 
https://github.com/university-of-york/sys-aws-layer-ruby-pg/pull/2

I wonder if there is a common script we can use to build these in the longer term. Maybe make all these build from a common repo.

But for now LGTM. :)

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
- && yum clean all \
- && rm -rf /var/cache/yum
-
 # Don't complain if bundler is run as root
 ENV BUNDLE_SILENCE_ROOT_WARNING=1
 
-# Install GitHub CLI for actions
-RUN yum install -y yum-utils \
- && yum-config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo \
- && yum install -y gh \
```
Comment (njm506) [inline ci/aws-lambda-ruby-dev/3.2/Dockerfile:14]: Is this included by something else now?

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,24 @@
 FROM public.ecr.aws/lambda/ruby:3.2
 
-# Install tools used when installing / building gems
-RUN yum install -y postgresql-devel sqlite-devel openssl-devel wget make gcc gcc-c++ git \
- && yum clean all \
- && rm -rf /var/cache/yum
```
Comment (njm506) [inline ci/aws-lambda-ruby-dev/3.2/Dockerfile:6]: We included these (within an individual step) to minimize image size after each yum step.

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile (hunk 2 above)
Comment (danmilleruk) [reply ci/aws-lambda-ruby-dev/3.2/Dockerfile:14]: Apologies, missed that one!

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile (hunk 3 above)
Comment (danmilleruk) [reply ci/aws-lambda-ruby-dev/3.2/Dockerfile:6]: Concatenated now 🙂

Comment (choffee) [reply ci/aws-lambda-ruby-dev/3.2/Dockerfile:6]: Still missing the clean up for a smaller image?

Comment (danmilleruk) [reply ci/aws-lambda-ruby-dev/3.2/Dockerfile:6]: 🤦 Added now

File: ci/aws-lambda-ruby-dev/3.2/Dockerfile (hunk 1 above)
Comment (danmilleruk) [reply ci/aws-lambda-ruby-dev/3.2/Dockerfile]: Fixed

**Review by njm506 — APPROVED** [review]

**Pull request merged**